augur/curate/apply_geolocation_rules.py

```diff
@@ -0,0 +1,222 @@
+"""
+Applies user curated geolocation rules to the geolocation fields.
+"""
+from collections import defaultdict
+from augur.errors import AugurError
+from augur.io.print import print_err
+from augur.utils import first_line
+
+
+class CyclicGeolocationRulesError(AugurError):
+    pass
+
+
+def load_geolocation_rules(geolocation_rules_file):
+    """
+    Loads the geolocation rules from the provided *geolocation_rules_file*.
+    Returns the rules as a dict:
+    {
+        regions: {
+            countries: {
+                divisions: {
+                    locations: corrected_geolocations_tuple
+                }
+            }
+        }
+    }
+    """
+    geolocation_rules = defaultdict(lambda: defaultdict(lambda: defaultdict(dict)))
+    with open(geolocation_rules_file, 'r') as rules_fh:
```

Comment:
i'm gonna have a bunch of silly questions i think: is there a pattern in augur around who is responsible for handling (or, i guess, not handling) exceptions?

Reply:
Yes, see #642.

Keep the questions coming! It's good to know what seems weird from a fresh perspective. Also this comment prompted me go back and update that outdated issue.